compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -804,6 +804,24 @@ abstract class PrepJSInterop extends plugins.PluginComponent
             reporter.error(tree.pos,
                 "@JSBracketAccess methods with two parameters must return Unit")
           }
+
+          val flatParams = sym.paramss.flatten
+
+          for {
+            param <- flatParams
+            if isScalaRepeatedParamType(param.tpe)
+          } {
+            reporter.error(param.pos,
+                "@JSBracketAccess methods may not have repeated parameters")
+          }
+
+          for {
+            firstParam <- flatParams.headOption
+            if firstParam.isParamWithDefault
+          } {
+            reporter.error(firstParam.pos, "The first parameter of a " +
+                "@JSBracketAccess method may not be a default parameter")
```

Comment:
> Seriously, that works?

Yes... It's all post JS argument extraction magic. So even this works:

```scala
@JSBracketAccess
def foo(x: Int*): Unit = js.native

x.foo()          // crash
x.foo(1)         // x[1]
x.foo(1, 2)      // x[1] = 2
x.foo(1, 2, 3)   // crash
x.foo(mySeq: _*) // crash
```

So if we make the default argument thingy just a warning, we also need to make this a warning and proper errors at the call sites (so we don't crash).

Do you think we could consider breaking backwards compatibility just a bit here?